test/rns.subdomains.test.ts

```diff
@@ -0,0 +1,198 @@
+    test('mainnet', async () => {
+      const web3 = new Web3(PUBLIC_NODE_MAINNET);
+      const rns = new RNS(web3);
+      await asyncExpectThrowError(async () => await rns.checkSubdomainAvailability('domain.rsk', ''), INVALID_LABEL);
+    });
+
+    test('testnet', async () => {
+      const web3 = new Web3(PUBLIC_NODE_TESTNET);
+      const rns = new RNS(web3);
+      await asyncExpectThrowError(async () => await rns.checkSubdomainAvailability('domain.rsk', ''), INVALID_LABEL);
+    });
+  });
+
+  describe('should fail when sending label with upper case characters', () => {
+    test('mainnet', async () => {
+      const web3 = new Web3(PUBLIC_NODE_MAINNET);
+      const rns = new RNS(web3);
+      await asyncExpectThrowError(async () => await rns.checkSubdomainAvailability('domain.rsk', 'iNVAlid'), INVALID_LABEL);
+    });
+
+    test('testnet', async () => {
+      const web3 = new Web3(PUBLIC_NODE_TESTNET);
+      const rns = new RNS(web3);
+      await asyncExpectThrowError(async () => await rns.checkSubdomainAvailability('domain.rsk', 'iNVAlid'), INVALID_LABEL);
+    });
+  });
+
+  describe('should fail when sending label with invalid characters', () => {
+    test('mainnet', async () => {
+      const web3 = new Web3(PUBLIC_NODE_MAINNET);
+      const rns = new RNS(web3);
+      await asyncExpectThrowError(async () => await rns.checkSubdomainAvailability('domain.rsk', 'iNVA-lid'), INVALID_LABEL);
+    });
+
+    test('testnet', async () => {
+      const web3 = new Web3(PUBLIC_NODE_TESTNET);
+      const rns = new RNS(web3);
+      await asyncExpectThrowError(async () => await rns.checkSubdomainAvailability('domain.rsk', 'iNVA-lid'), INVALID_LABEL);
+    });
+  });
+});
+
+describe('checkSubdomainAvailability happy paths', () => {
+  describe('should return false if label is not available', () => {
+    test('mainnet', async () => {
+      const web3 = new Web3(PUBLIC_NODE_MAINNET);
+      const rns = new RNS(web3);
+      const available = await rns.checkSubdomainAvailability('testing.rsk', 'multichain');
+      expect(available).toBe(false);
+    });
+
+    test('testnet', async () => {
+      const web3 = new Web3(PUBLIC_NODE_TESTNET);
+      const rns = new RNS(web3);
+      const available = await rns.checkSubdomainAvailability('testing.rsk', 'multichain');
+      expect(available).toBe(false);
+    });
+  });
+
+  describe('should return true if label is available', () => {
+    test('mainnet', async () => {
+      const web3 = new Web3(PUBLIC_NODE_MAINNET);
+      const rns = new RNS(web3);
+      const available = await rns.checkSubdomainAvailability('testing.rsk', 'available');
+      expect(available).toBe(true);
+    });
+
+    test('testnet', async () => {
+      const web3 = new Web3(PUBLIC_NODE_TESTNET);
+      const rns = new RNS(web3);
+      const available = await rns.checkSubdomainAvailability('testing.rsk', 'available');
+      expect(available).toBe(true);
+    });
+  });
+});
```

Comment:
Excellent testing! Maybe we can refactor using `afterEach` for the assertions. It will be very useful when integrating local tests.